src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -216,6 +216,7 @@ public InnReachResponseDTO cancelPatronHold(String trackingId, String centralCod
 
     removeItemTransactionInfo(itemId)
       .ifPresent(this::removeHoldingsTransactionInfo);
+    clearPatronAndItemInfo(transaction.getHold());
 
     log.info("Item request successfully cancelled");
 
```

Comment:
persist new transaction state before cancelling request and updating item. Otherwise the kafka event will be processed before the new state is saved and it will lead to incorrect processing.

`      transaction = saveInNewDbTransaction(transaction);
`

Reply:
The issue is similar to https://issues.folio.org/browse/MODINREACH-257